File: airesources/Java/hlt/Metadata.java
package hlt;

public class Metadata {
    final private String[] metadata;
    private int index = 0;

    public Metadata(final String[] metadata) {
        this.metadata = metadata;
    }

    public String pop() {
        return metadata[index++];
    }

    public boolean isEmpty() {
        return index == metadata.length;
    }
}

File: airesources/CSharp/Halite2/hlt/Metadata.cs
namespace Halite2.hlt
{
    public class Metadata
    {
        private string[] metadata;
        private int index = 0;

        public Metadata(string[] metadata)
        {
            this.metadata = metadata;
        }

        public string Pop()
        {
            return metadata[index++];
        }

        public bool IsEmpty()
        {
            return index == metadata.Length;
        }
    }
}
